### PR TITLE
[MOB-1788] Clear in-app messages on logout

### PR DESCRIPTION
**Pull request by vbabenkoru**
Make sure we clear the in-app queue immediately on logout.

### CODE REVIEW

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java [hunk 1]
```diff
@@ -1328,13 +1328,22 @@ private void retrieveEmailAndUserId() {
...
+        if (config.autoPushRegistration) {
             disablePush();
         }
+        getInAppManager().reset();
     }
 
     private void onLogIn() {
-        if (config.autoPushRegistration && isInitialized()) {
+        if (!isInitialized()) {
+            return;
```
Comment (Ayyanchira) [inline iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java:1343]: Even after login if we haven't initialized, shouldn’t we attempt to initialize??

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppManager.java [hunk 2]
```diff
@@ -167,6 +167,20 @@ public void execute(String payload) {
         });
     }
 
+    /**
+     * Clear all in-app messages.
+     * Should be called on user logout.
+     */
```
Comment (Ayyanchira) [inline iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppManager.java:173]: Feels like reset name should be for something which is resetting everything...
May be more specific name like clearAllMessagesFromLocalStorage()

File: iterableapi/src/test/java/com/iterable/iterableapi/IterableInAppManagerTest.java [hunk 3]
```diff
@@ -150,6 +150,20 @@ public void testSyncInApp() throws Exception {
...
+        dispatcher.enqueueResponse("/inApp/getMessages", new MockResponse().setBody(IterableTestUtils.getResourceString("inapp_payload_single.json")));
+        IterableInAppManager inAppManager = IterableApi.getInstance().getInAppManager();
+        assertEquals(0, inAppManager.getMessages().size());
+
+        inAppManager.syncInApp();
+        Robolectric.flushBackgroundThreadScheduler();
+        Robolectric.flushForegroundThreadScheduler();
+        assertEquals(1, inAppManager.getMessages().size());
+        inAppManager.reset();
+        assertEquals(0, inAppManager.getMessages().size());
```
Comment (Ayyanchira) [inline iterableapi/src/test/java/com/iterable/iterableapi/IterableInAppManagerTest.java:164]: 👍

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java (hunk 1 above)
Comment (vbabenkoru) [reply iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java:1343]: Here is the code for `isInitialized()`:
```
    private boolean isInitialized() {
        return _apiKey != null && (_email != null || _userId != null);
    }
```
Basically, if there is no email or userId, there is no user to operate with.

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppManager.java (hunk 2 above)
Comment (vbabenkoru) [reply iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppManager.java:173]: I'm mirroring the name in iOS for consistency.. Technically it does reset everything in the in-app manager.

**Pull request merged**